runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -185,6 +194,10 @@ public void write(IToken token, Object data) throws OverwriteException {
         return addressesMap;
     }
 
+    public Map<Long, ILogData> read(Iterable<Long> addresses) {
```

Comment:
This function does not guarantee ordering. The map returned by the cache is an ImmutableMap that references an HashMap. The HashMap is created in Caffeine code, so we have a bit our hands tight here. Maybe we can create a TreeMap out of the map returned by readCache.getAll ?